### PR TITLE
Test sprite atlas

### PR DESCRIPTION
**Pull request by emeline-berenguier**
### Description

Add Tests for the SpriteAtlas empty Descriptor

### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
- [x] Docs for new/changed API's.
    - Code is annotated using Xmldoc syntax.

### CODE REVIEW

File: Editor/TestUtils/TestAsset.cs [hunk 1]
```diff
@@ -54,6 +54,29 @@ public static TestAsset Save(UnityEngine.Object asset, string fileName)
             return tempAsset;
         }
 
+#if UNITY_2020_1_OR_NEWER
+        //SpriteAtlasAsset Save is not compatible with the AssetDatabase save
+        //Alternative function to create a TestAsset from a SpriteAtlas
+        public static TestAsset SaveSpriteAtlasAsset(SpriteAtlasAsset asset, string fileName)
+        {
+            var tempAsset = new TestAsset(fileName);
+            #if UNITY_2021_1_OR_NEWER
```
Comment (unitystevem) [inline Editor/TestUtils/TestAsset.cs:63]: The formatting is a bit weird here. I'd normally expect #if/else/endif statements to be fully left-aligned

**Review by unitystevem — APPROVED** [review]
Would be nice to fix the alignment issue on the preprocessor statements in TestAsset.cs, but apart from that this looks good to go.
